language-support/hs/bindings/examples/nim-console/src/NimLedger.hs

```diff
@@ -34,7 +34,7 @@ connect :: Logger -> IO Handle
 connect log = do
     lid <- run 5 getLedgerIdentity
     ids <- run 5 $ listPackages lid
-    [pid,_,_] <- return ids -- assume Nim stuff is in the 1st of 3 packages
+    [_,pid,_] <- return ids -- guess which is the Nim package -- TODO: fix this properly!
```

Comment:
The fix should be quite easy given that I already added the logic to the tests so I would suggest that you just fix it properly in this PR.

Reply:
I don't think your fix which discovers the mainPackageId for the tests is directly applicable to the nim/chat apps because it starts from the dar, whereas for the apps, the dar is not known/available, and so we need to actually traverse into the package contents to find which package provides the module we are looking for.